linalg/solvers.hpp

```diff
@@ -699,15 +699,26 @@ class LBFGSSolver : public NewtonSolver
 {
 protected:
    int m = 10;
+   mutable Array<Vector *> skArray, ykArray;
```

Comment:
I think you should redefine `SetOperator()` and call `InitializeStorageVectors()` there, since that's when the `width` becomes available. For example:
```
virtual SetOperator(op)
{
NewtonSolver::SetOperator(op)
InitializeStorageVectors()
}
```
Then the calls to `InitializeStorageVectors` in the constructors won't be needed (they don't know the correct `width` anyway).